### PR TITLE
feat(sca): licenses suppressions by type

### PR DESCRIPTION
**Pull request by NoaAzoulay**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

[//]: # "
    # PR Title
    Be aware that we use the title to create changelog automatically and therefore only allow specific prefixes
    - break:    to indicate a breaking change, this supersedes any of the types
    - feat:     to indicate new features or checks
    - fix:      to indicate a bugfix or handling of edge cases of existing checks
    - docs:     to indicate an update to our documentation
    - chore:    to indicate adjustments to workflow files or dependency updates
    - platform: to indicate a change needed for the platform
    Additionally a scope is needs to be added to the prefix, which indicates the targeted framework, in doubt choose 'general'.
    ex.
    feat(terraform): add CKV_AWS_123 to ensure that VPC Endpoint Service is configured for Manual Acceptance
"

## Description
added suppressions for licenses by type (name)
UTs for suppressions by policy + by type 
## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my feature, policy, or fix is effective and works
- [x] New and existing tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: checkov/common/bridgecrew/integration_features/features/suppressions_integration.py [hunk 1]
```diff
@@ -154,7 +154,7 @@ def _check_suppression(self, record: Record, suppression: dict[str, Any]) -> boo
             return False
 
         elif type == 'Cves':
-            if self.bc_integration.repo_id in suppression['accountIds'] and record.file_abs_path == suppression['cves'][0]['id'][1:]:
+            if self.bc_integration.repo_id in suppression['accountIds'] and record.file_abs_path.endswith(suppression['cves'][0]['id'][1:]):
```
Comment (itai1357) [inline checkov/common/bridgecrew/integration_features/features/suppressions_integration.py:157]: does `suppression['cves'][0]['id'][1:]` represent a file name?

File: checkov/common/bridgecrew/integration_features/features/suppressions_integration.py [hunk 2]
```diff
@@ -160,6 +160,12 @@ def _check_suppression(self, record: Record, suppression: dict[str, Any]) -> boo
                         return True
             return False
 
+        elif type == 'LicenseType':
+            for license_type in suppression['licenseTypes']:
+                if record.vulnerability_details and record.vulnerability_details['license'] == license_type:
+                    return True
+            return False
```
Comment (Saarett) [inline checkov/common/bridgecrew/integration_features/features/suppressions_integration.py:167]: This code looks like this
https://docs.python.org/3/library/functions.html#any
WDYT about changing it? :)

File: checkov/sca_package/output.py [hunk 3]
```diff
@@ -124,6 +124,8 @@ def create_cli_output(fixable: bool = True, *cve_records: list[Record]) -> str:
                         }
                     )
                 elif record.check_name == SCA_LICENSE_CHECK_NAME:
+                    if record.check_result["result"] == CheckResult.SKIPPED:
+                        continue
```
Comment (Saarett) [inline checkov/sca_package/output.py:128]: 👍

File: checkov/common/bridgecrew/integration_features/features/suppressions_integration.py [hunk 4]
```diff
@@ -160,6 +160,12 @@ def _check_suppression(self, record: Record, suppression: dict[str, Any]) -> boo
                         return True
             return False
 
+        elif type == 'LicenseType':
+            for license_type in suppression['licenseTypes']:
```
Comment (itai1357) [inline checkov/common/bridgecrew/integration_features/features/suppressions_integration.py:164]: I see in the prev types ('Cves' 'CvesAccounts') there are check for `if self.bc_integration.repo_id in suppression['accountIds']`
are you sure we don't need it in our case?

**Review by Saarett — COMMENTED** [review]
Great job! Added 1 comment :)

File: tests/common/integration_features/test_suppressions_integration.py [hunk 5]
```diff
@@ -628,6 +628,77 @@ def test_supress_by_cve_for_image_scan_with_different_repo_id(self):
         self.assertFalse(suppressions_integration._check_suppression(record3, suppression))
         self.assertFalse(suppressions_integration._check_suppression(record4, suppression))
 
+    def test_suppress_licenses_by_policy(self):
```
Comment (itai1357) [inline tests/common/integration_features/test_suppressions_integration.py:631]: nice!
I guess suppression by policy is already supported?

Comment (NoaAzoulay) [reply tests/common/integration_features/test_suppressions_integration.py:631]: yes

Comment (NoaAzoulay) [reply tests/common/integration_features/test_suppressions_integration.py:631]: right :)

File: checkov/common/bridgecrew/integration_features/features/suppressions_integration.py (hunk 4 above)
Comment (itai1357) [reply checkov/common/bridgecrew/integration_features/features/suppressions_integration.py]: cool

**Review by itai1357 — APPROVED** [review]

**Review by ayajbara — APPROVED** [review]

**Pull request merged**